src/main/java/edu/kis/powp/jobs2d/drivers/monitoring/DriverMonitorDecorator.java

```diff
@@ -0,0 +1,29 @@
+package edu.kis.powp.jobs2d.drivers.monitoring;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class DriverMonitorDecorator implements Job2dDriver {
+    private final Job2dDriver driver;
+    private final DriverUsageMonitor monitor;
+    private final DriverMonitor outputMonitor;
+
+    public DriverMonitorDecorator(Job2dDriver driver, DriverUsageMonitor monitor, DriverMonitor outputMonitor) {
```

Comment:
A full observer pattern implementation would be more suitable for OCP in this case (typical observer usage).